sympy/sets/sets.py

```diff
@@ -216,13 +216,11 @@ def _complement(self, other):
             return S.EmptySet
 
         elif isinstance(other, FiniteSet):
-            both_l = non_num, num = [], []
-            for el in other:
-                both_l[el.is_number].append(el)
-            non_num = FiniteSet(*non_num)
-            out_range = FiniteSet(*[el for el in num if self.contains(el) == False])
-            if non_num:
-                return Union(out_range, Complement(non_num, self, evaluate = False))
+
+            none_range = FiniteSet(*[el for el in other if self.contains(el) not in [True, False]])
```

Comment:
Or, if you import sift from utilities:

```python
sifted = sift(other, lambda i: self.contains(i))
# ignore those that are contained in self
return Union(sifted[True],
    Complement(sifted[None], self, evaluate=False) if sifted[None] else
    S.EmptySet)
```

Reply:
If you choose not to do this, that's ok. But please close up the space around `=` as I did in the snippet above.

Reply:
@smichr 
```python
sifted = sift(other, lambda i: self.contains(i))
# ignore those that are contained in self
return Union(sifted[True],
    Complement(sifted[None], self, evaluate=False) if sifted[None] else
    S.EmptySet)
```
I think it should be `return Union(sifted[False]...` , but then also it is giving some errors . 
```
raise TypeError("Input must be Sets or iterables of Sets")
TypeError: Input must be Sets or iterables of Sets
```
As sift is returning a list .